core/src/main/java/org/bitcoinj/core/LockTime.java

```diff
@@ -127,18 +161,16 @@ public int blockHeight() {
      * @return true if specified as a timestamp
      */
     public boolean isTimestamp() {
-        return value >= THRESHOLD;
+        return this instanceof TimeLock;
     }
 
     /**
      * Gets the lock time as a timestamp.
      * @return lock time as a timestamp
      * @throws IllegalStateException if the lock time is not specified as a timestamp
      */
-    public Instant timestamp() {
-        if (!isTimestamp())
-            throw new IllegalStateException("lock time not specified as a timestamp");
-        return Instant.ofEpochSecond(value);
+    public Optional<Instant> getTimestamp() {
```

Comment:
`timestamp()`?

Reply:
Again, we don't want to conflict with the method in the subclass.  So the `get` implies some level of indirection.

Reply:
... and of course we could remove this altogether.

Reply:
Removed.